### DISCUSSION TITLE
Is it possible to omit the consent redirection?

### DISCUSSION THREAD

**Discussion by Wayne82** [ory/hydra#2562] [Q&A] [answered]
I am trying hydra to build the OAuth flow for my own application, and I am wondering whether I can omit the consent redirection (not just skip the UI) at all.

And thanks for this awesome project!

Regards,
Wayne

**Comment by vinckr** [comment, marked as answer]
Hey Wayne, 
I am not sure if it is possible at all but it would break Open ID Connect certification. 
Afaik the [Resource Owner Password Credentials Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-4.3) does not require a callback URL, but that has been deprecated for security reasons (in short: it is very insecure).

I think the [JSON Web Token (JWT) Profile for OAuth 2.0 Authorization Grant](https://datatracker.ietf.org/doc/html/rfc7523) could solve your problem.

There is a draft PR going to add this to Hydra, check it out: https://github.com/ory/hydra/pull/2384 .

**Reply by Wayne82** [reply]
Thanks for your reply @vinckr. 
I am not quite sure about the consent flow, but is it related to which OAuth grant type I choose? I thought I should be able to skip the consent flow even using the authorization code grant type.
I checked auth0 (https://auth0.com/), which has a setting "Allow Skipping User Consent" for the applications flagged as **First Party**. So, does hydra have similar functions/settings?

**Reply by Benehiko** [reply]
Hi @Wayne82 

I believe this is what you're looking for?

> The response (see below in "Consent Challenge Response" tab) contains information about the consent request. The body contains a skip value. If the value is false, the user interface must be shown. If skip is true, you should not show the user interface but instead just accept or reject the consent request! 

https://www.ory.sh/hydra/docs/concepts/consent#the-consent-endpoint

**Reply by Wayne82** [reply]
Thanks for the reference @Benehiko. 
Actually, I expect to even skip the consent redirection flow. After accepting the login, let hydra redirect to my callback URL with the authorization code directly (no redirect to consent URL at all)
Is it possible?

**Reply by Benehiko** [reply]
Hey @Wayne82 

I don't think there is a complete redirect skip with a configuration change on hydra. The only way I can think of (and also a bit hacky) would be to do all of the consent requests on the server side once the login request has been completed. But this would mean that you would have to write some server-side software to make the consent request to hydra on behalf of the client app. 
This would require you to initialise the consent request, store its cookies and consent_challenge and then accept the request. Then you would also need to return the auth code to the callback URL by yourself. In a sense your service would have to be a reverse proxy layered on top of hydra. 

This is extremely hacky since you're preventing the flow from completing in its intended way. I am also not sure if this would have security implications.

**Reply by aeneasr** [reply]
As @Benehiko said you can not skip the redirection but what you can do is to accept the second redirection immediately. The user will probably not notice as this can be done in < 100ms.

@vinckr this question is coming up probably once a month - could you maybe add an item to the "guides" section of hydra? :)
